src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -74,22 +73,36 @@ private void processZipFile(CloudBlobContainer container, String zipFilename)
         log.info("Processing zip file {}", zipFilename);
 
         CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(zipFilename);
-        BlobInputStream blobInputStream = cloudBlockBlob.openInputStream();
-
-        //Zip file will include metadata.json and collection of pdf documents
-        try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
-            ZipFileProcessor zipFileProcessor = processZipInputStream(zis, zipFilename, container.getName());
-
-            if (zipFileProcessor != null) {
-                processParsedEnvelopeDocuments(
-                    zipFileProcessor.getEnvelope(),
-                    zipFileProcessor.getPdfs(),
-                    cloudBlockBlob
-                );
+
+        CloudBlockBlob blobWithLeaseAcquired = acquireLease(cloudBlockBlob, container.getName(), zipFilename);
+
+        if (Objects.nonNull(blobWithLeaseAcquired)) {
+            BlobInputStream blobInputStream = blobWithLeaseAcquired.openInputStream();
+
+            //Zip file will include metadata.json and collection of pdf documents
+            try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
+                ZipFileProcessor zipFileProcessor = processZipInputStream(zis, zipFilename, container.getName());
+
+                if (zipFileProcessor != null) {
+                    processParsedEnvelopeDocuments(
+                        zipFileProcessor.getEnvelope(),
+                        zipFileProcessor.getPdfs(),
+                        blobWithLeaseAcquired
+                    );
+                }
             }
         }
     }
 
+    private CloudBlockBlob acquireLease(CloudBlockBlob cloudBlockBlob, String containerName, String zipFilename) {
+        // TODO - Lease time  needs to be made configurable
+        return errorWrapper.wrapAcquireLeaseFailure(containerName, zipFilename, () -> {
+            //The value of the parameter 'leaseTimeInSeconds' should be between 15 and 60 seconds.
+            cloudBlockBlob.acquireLease(15, null);
```

Comment:
instead of blindly trying to lease and throwing an exception, can we check if the files is already locked and only then acquire a lease for it?
Or, even better, acquire lease for first available file?